istioctl/cmd/istioctl/proxystatus.go

```diff
@@ -52,7 +52,7 @@ Retrieves last sent and last acknowledged xDS sync from Pilot to each Envoy in t
 				if err != nil {
 					return err
 				}
-				path = fmt.Sprintf("/debug/config_dump?proxyID=%v", args[0])
+				path = fmt.Sprintf("/debug/config_dump?proxyID=%s.%s", podName, ns)
```

Comment:
This is not quite the correct behaviour. Similar to `proxy-config` it should be able to handle both `proxy.namespace` and `proxy -n namespace`. You can probably even re-use the code.

Reply:
@liamawhite Yes, this simple change can work for both proxy.namespace and  proxy -n namespace as the namespace handling logic is the in handleNamespace() which is the same with proxy-config already.  Below are two sample output with this fix:
```
istioctl proxy-status istio-ingressgateway-5979b99885-h7t85.istio-system

Clusters Match
Listeners Match
Routes Match

```

```
istioctl proxy-status istio-ingressgateway-5979b99885-h7t85 -n istio-system

Clusters Match
Listeners Match
Routes Match
```

Reply:
Ah, forgot that was handled there. Probably should have spent more time checking but was in a rush 🙂.

Could you add some test cases to the existing unit tests to cover these?

Reply:
Have added two test cases, one is podName.ns and the other one is -n ns